This patch came from http://sarah-a-happy.livejournal.com/90345.html
It makes compatability changes for changed function names and a macro change
in kernel 2.6.25.
--- build_mod/firegl_public.c.orig     2008-04-06 01:20
:21.000000000 -0300
+++ build_mod/firegl_public.c  2008-04-06 01:17:58.0000
00000 -0300
@@ -1885,7 +1885,7 @@
  */
 KCL_TYPE_Cap ATI_API_CALL KCL_PosixSecurityCapGetEffectiveVector()
 {
-    return cap_t(current->cap_effective);
+    return (current->cap_effective).cap[0];
 }
 
 /** \brief Set vector of effective security caps for the current process
@@ -1893,7 +1893,7 @@
  */
 void ATI_API_CALL KCL_PosixSecurityCapSetEffectiveVector(KCL_TYPE_Cap cap)
 {
-    cap_t(current->cap_effective) = cap;
+    (current->cap_effective).cap[0] = cap;
 }
 
 /** \brief Get number of available RAM pages
--- build_mod/firegl_public.c.orig     2008-04-06 02:10
:13.000000000 -0300
+++ build_mod/firegl_public.c  2008-04-06 02:23:05.0000
00000 -0300
@@ -120,6 +120,7 @@
 #include <asm/cpufeature.h>
 #ifdef CONFIG_MTRR
 #include <asm/mtrr.h>
+#include <asm/cacheflush.h>
 #endif
 #include <asm/delay.h>
 #include <linux/agp_backend.h>
@@ -1158,7 +1159,7 @@
 
 int ATI_API_CALL KCL_SetPageNoCache(void* virt, int pages)
 {
-    return change_page_attr(virt_to_page(virt), pages, PAGE_KERNEL_NOCACHE);
+    return set_pages_uc(virt_to_page(virt), pages);
 }
 
 /** /brief Call global kernel task/thread scheduler */
